tools/ktx/command_extract.cpp

```diff
@@ -23,38 +23,150 @@
+    }
+    friend bool operator!=(const Selector& var, All_t) {
+        return !var.all;
+    }
+    friend bool operator>=(const Selector& var, uint32_t value) {
+        return !var.all && var.value >= value;
+    }
+    friend bool operator>=(uint32_t value, const Selector& var) {
+        return !var.all && value >= var.value;
+    }
+};
+
+static uint32_t value_or(const std::optional<Selector> opt, uint32_t fallback) {
+    return opt ? opt->value : fallback;
+}
+
+} // namespace ktx
+
+template<> struct fmt::formatter<ktx::Selector> : fmt::formatter<uint32_t> {
+    template <typename FormatContext>
+    auto format(const ktx::Selector& var, FormatContext& ctx) const -> decltype(ctx.out()) {
+        return var.all ?
+            formatter<std::string_view>{}.format("all", ctx) :
+            formatter<uint32_t>::format(var.value, ctx);
+    }
+};
+
+namespace ktx {
+
 // -------------------------------------------------------------------------------------------------
 
 /** @page ktxtools_extract ktx extract
 @~English
 
-Extracts a KTX2 file.
-
-@warning TODO Tools P5: This page is incomplete
+Extract selected images from a KTX2 file.
 
 @section ktxtools_extract_synopsis SYNOPSIS
-    ktx extract [options] @e input_file
+    ktx extract [option...] @e input_file @e output_path
 
 @section ktxtools_extract_description DESCRIPTION
+    @b ktx @b extract can extract one or multiple images from the KTX2 file specified as the
+    @e input_file argument and based on the format save them as RAW, EXR or PNG image files
+    to the @e output_path.
+    If the input file is invalid the first encountered validation error is displayed
+    to the stderr and the command exits with the relevant non-zero status code.
+
+    The @e output_path is interpreted as output filepath for single and output directory for
+    multi-image extracts.
+    When extracting multiple images with either '--all' or any of the 'all' args the
+    following naming is used for each output file:
+    <pre>output_path/output_level{}_face{}_layer{}_depth{}.extension</pre>
+    - Where the @e _level{} part is only present if the source texture has more than 1 level
+    - Where the @e _face{} part is only present if the source texture is cubemap or cubemap array (Cubemap)
+    - Where the @e _layer{} part is only present if the source texture is an array texture (Array)
+    - Where the @e _depth{} part is only present if the source texture baseDepth is more than 1 (3D)
+    - Where the @e {} is replaced with the numeric index of the given component starting from 0
+    - Where the @e extension part is "raw", "png" or "exr" based on the export format<br />
+    Note: The inclusion of the optional parts are determined by the source texture regardless of
+    which images are requested.
+
+    For non-raw exports the output image format is chosen to be the smallest related lossless
+    format:
+    - _UNORM formats exported as PNG 8 or 16 bit (except for Depth/Stencil)
```

Comment:
What about depth/stencil? Explain their treatment. Are they not exported?

Reply:
The complete support for depth/stencil formats were not part of this PR (as well as the other formats which needed extra work).

I have made an attempt to reword the rules:
- _UNORM formats (except Depth/Stencil) exported as PNG with RGB/RGBA 8/16 bit
- _SINT/_UINT formats exported as EXR with R/RG/RGB/RGBA Half/Float/UInt
- _SFLOAT/_UFLOAT formats exported as EXR with R/RG/RGB/RGBA Half/Float/UInt
- D16_UNORM exported as PNG with luminance (Gray) 16 bit
- Other Depth/Stencil formats exported as EXR with D/S/DS Half/Float

Not sure on what would be the best format to express these rules or weather we should just include the whole "format conversion table" or not.
I am not happy about this wording either. I am open to suggestions.

Reply:
I think this wording is okay together with the preceding sentence (not shown here) about choosing the smallest related format.